pkg/kapp/app/recorded_app_changes.go

```diff
@@ -17,22 +17,24 @@ import (
 )
 
 const (
-	isChangeLabelKey   = "kapp.k14s.io/is-app-change"
-	isChangeLabelValue = ""
-	changeLabelKey     = "kapp.k14s.io/app-change-app" // holds app name or label
+	isChangeLabelKey     = "kapp.k14s.io/is-app-change"
+	isChangeLabelValue   = ""
+	legacyChangeLabelKey = "kapp.k14s.io/app-change-app"       // holds app name
+	changeLabelKey       = "kapp.k14s.io/app-change-app-label" // holds app label
 )
 
 type RecordedAppChanges struct {
-	nsName  string
-	appName string
+	nsName           string
+	appName          string
+	changeLabelValue string
 
-	appLabelValue string
+	appChangeUsesAppLabel bool
 
 	coreClient kubernetes.Interface
 }
 
-func NewRecordedAppChanges(nsName, appName, appLabelValue string, coreClient kubernetes.Interface) RecordedAppChanges {
-	return RecordedAppChanges{nsName, appName, appLabelValue, coreClient}
+func NewRecordedAppChanges(nsName, appName, changeLabelValue string, appChangeUsesAppLabel bool, coreClient kubernetes.Interface) RecordedAppChanges {
```

Comment:
```suggestion
func NewRecordedAppChanges(nsName, appName, appChangeLabelValue string, appChangeUsesAppLabel bool, coreClient kubernetes.Interface) RecordedAppChanges {
```

Reply:
+1

Reply:
(I kept it `changeLabelValue` to be consistent with `changeLabelKey`)
Do you want me to make this change before merging?